execution_engine/src/core/runtime_context/mod.rs

```diff
@@ -1118,4 +1132,32 @@ where
         self.metered_write_gs_unsafe(dictionary_key, wrapped_cl_value)?;
         Ok(())
     }
+
+    pub fn system_contracts(&self) -> Vec<ContractHash> {
+        self.tracking_copy
+            .borrow_mut()
+            .get_system_contracts(self.correlation_id)
+            .expect("should have system contract registry")
+            .values()
+            .cloned()
+            .collect()
+    }
+
+    pub fn get_system_contract(&self, name: &str) -> ContractHash {
+        *self
+            .tracking_copy
+            .borrow_mut()
+            .get_system_contracts(self.correlation_id)
+            .map_err(Error::from)
+            .expect("should have system contracts registry")
+            .get(name)
+            .expect("should have system contract registry")
+    }
```

Comment:
This seems particularly fragile to me.  Let's remove the ability to pass arbitrary strings through here and have 4 separate getters, one per system contract.  Each should be `pub(super)` ideally or `pub(crate)` if really required.  They should be named like `mint_contract_hash(&self) -> ContractHash`.

I'd be inclined to add those as methods on the tracking copy itself and just delegate here, but if that causes problems, leave it for now.